File: src/lib/entry/command/create-log-command.ts
import { CommandName } from "../../globals"
import JSONCommandType, { JSONCommandTypeArgs } from "./command-type/json-command-type"

const COMMAND_NAME_BYTE = new Uint8Array([CommandName.CREATE_LOG])

export default class CreateLogCommand extends JSONCommandType {
    constructor(args: JSONCommandTypeArgs) {
        if (!args.commandNameU8) {
            args.commandNameU8 = COMMAND_NAME_BYTE
        }
        super(args)
    }
}

File: external/logsrdcpp/src/entry/command/CreateLogCommand.cpp
#include "CreateLogCommand.h"

namespace logsrd {

CreateLogCommand::CreateLogCommand(JSONCommandTypeArgs args)
    : JSONCommandType([&]() {
        if (args.commandNameU8.empty()) {
            args.commandNameU8 = std::vector<uint8_t>{COMMAND_NAME_BYTE};
        }
        return args;
    }())
{}

} // namespace logsrd
